validators/validator_set.go

```diff
@@ -298,6 +298,11 @@ func (t *Topology) applyPromotion(performanceScore, rankingScore map[string]num.
 		}
 	}
 
+	// demoted tendermint validators are also ersatz so we need to add them
+	for _, id := range demotedFromTM {
+		ersatzValidators = append(ersatzValidators, t.validators[id])
+	}
+
 	// demote from ersatz to pending due to more ersatz than slots allowed
 	ersatzValidators, waitingListValidators, _ = demoteDueToLackOfSlots(ersatzValidators, waitingListValidators, ValidatorStatusErsatz, ValidatorStatusPending, t.numberOfErsatzValidators, int64(t.currentBlockHeight+1))
```

Comment:
The one thing I am unsure of is whether we should be then removing the `demotedFromTM` validators from `ersatzValidators` before we call promote before to stop a TM -> pending. Its unclear to me from the spec whether in: 
"If a node that submitted the transaction to join and satisfies all other conditions and has a higher score than the lowest scoring ersatz validator (scaled up by the incumbent factor)"

`lowest scoring ersatz validator` means the lowest scoring before the tendermint <-> ersatz promotions are applied, or after the tendermint <-> ersatz promotions.